File: common/src/main/java/com/hartwig/actin/datamodel/clinical/PriorOtherCondition.java
package com.hartwig.actin.datamodel.clinical;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PriorOtherCondition {

    @NotNull
    public abstract String name();

    @NotNull
    public abstract Set<String> doids();

    @NotNull
    public abstract String category();

}

File: common/src/main/java/com/hartwig/actin/clinical/datamodel/PriorOtherCondition.kt
package com.hartwig.actin.clinical.datamodel

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class PriorOtherCondition {
    abstract fun name(): String
    abstract fun year(): Int?
    abstract fun month(): Int?
    abstract fun doids(): Set<String?>
    abstract fun category(): String

    @JvmField
    abstract val isContraindicationForTherapy: Boolean
}
